documentation/modules/upgrading/proc-downgrade-brokers-older-kafka.adoc

```diff
@@ -70,6 +51,11 @@ NOTE: You must format the value of `log.message.format.version` as a string to p
 +
 See xref:con-versions-and-images-str[]
 
+.. If you changed the Kafka listener configuration to use the array format of the `GenericKafkaListener` schema,
+revert the `Kafka.spec.kafka.listeners` to use the old format.
++
+See xref:con-upgrade-listeners-str[]
```

Comment:
This has to be part of the Cluster Operator downgrade. It does not have to be changed with Kafka downgrade. The operator supports the new format with any Kafka version it supports.

I also wonder if we should make it more generic: _Revert any changes to the CRs_ or something. As there are more new things in each release. E.g. rack awareness in Connect, host aliases and other things are not supported after the operator downgrade and need to be removed. So making it more generic would be IMHO a plus.

Reply:
Removed from this procedure and added a new generic procedure to the CO downgrade procedure.

![image](https://user-images.githubusercontent.com/47596553/100908609-98196400-34c3-11eb-8d32-0ff2c022d596.png)